### DISCUSSION TITLE
Java opens/exports and tycho configurator

### DISCUSSION THREAD

**Discussion by karypid** [eclipse-tycho/tycho#2873] [General]
Hello,

I've been using the [Tycho project configurators](https://github.com/tesla/m2eclipse-tycho/releases/download/latest/) to import maven projects into Eclipse.

I have come across a project that requires `--add-exports`  and defines:

```
                        <plugin>
                            <groupId>org.apache.maven.plugins</groupId>
                            <artifactId>maven-compiler-plugin</artifactId>
                            <configuration>
                                <source>${java.source.version}</source>
                                <target>${java.target.version}</target>
                                <compilerArgs>
                                    <arg>--add-exports</arg>
                                    <arg>java.base/sun.nio.cs=ALL-UNNAMED</arg>
                                    <arg>--add-exports</arg>
                                    <arg>java.base/jdk.internal.misc=ALL-UNNAMED</arg>
                                </compilerArgs>
                            </configuration>
                        </plugin>
```

Whenever I import this project, I have to manually go to the  project's properties and in the "Java Build Path" and define these in "Module Dependencies".

The annoying thing is that if I update the maven project, the settings get overwritten and I need to manually update the build path again.

Is there some way to define these in `pom.xml` which would allow the Tycho project configurator to pick them up automatically, or is this simply something I have to live with?

**Comment by laeubi** [comment]
@karypid please note that https://github.com/tesla/m2eclipse-tycho/

1. is not maintained by Tycho project
2. is not required to use Tycho with m2e

beside that it seems more a bug of m2e to not honor the JPMS settings on project update.

> Is there some way to define these in pom.xml which would allow the Tycho project configurator to pick them up automatically

I don't think so

> or is this simply something I have to live with?

Simply uninstall the Tycho project configurator and see if it works with plain m2e, if not report a bug there.

**Reply by karypid** [reply]
Yes, it seems to work fine with m2e. The relevant compiler args no "just work".
